contrib/download_cmake.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# Script to download newest version of cmake on linux (or mac)
+# saves you the trouble of compiling it if you don't have root
+set -e # stop on failure
+cd "$(dirname "$0")"/.. # run in top-level directory
+
+CMAKE_VERSION_MAJOR=3
+CMAKE_VERSION_MINOR=7
+CMAKE_VERSION_PATCH=1
+CMAKE_VERSION_MAJMIN=$CMAKE_VERSION_MAJOR.$CMAKE_VERSION_MINOR
+CMAKE_VERSION=$CMAKE_VERSION_MAJMIN.$CMAKE_VERSION_PATCH
+
+# listed at https://cmake.org/files/v$CMAKE_VERSION_MAJMIN/cmake-$CMAKE_VERSION-SHA-256.txt
+# for the files cmake-$CMAKE_VERSION-Darwin-x86_64.tar.gz
+# and cmake-$CMAKE_VERSION-Linux-x86_64.tar.gz
+CMAKE_SHA256_DARWIN=1851d1448964893fdc5a8c05863326119f397a3790e0c84c40b83499c7960267
+CMAKE_SHA256_LINUX=7b4b7a1d9f314f45722899c0521c261e4bfab4a6b532609e37fef391da6bade2
+
+PLATFORM="$(uname)-$(uname -m)"
+FULLNAME=cmake-$CMAKE_VERSION-$PLATFORM
+case $PLATFORM in
+  Darwin-x86_64)
+    CMAKE_SHA256=$CMAKE_SHA256_DARWIN
+    CMAKE_EXTRACTED_PATH=$FULLNAME/CMake.app/Contents/bin/cmake;;
+  Linux-x86_64)
+    CMAKE_SHA256=$CMAKE_SHA256_LINUX
+    CMAKE_EXTRACTED_PATH=$FULLNAME/bin/cmake;;
+  *)
+    echo "This script only supports x86_64 Mac and Linux. For other platforms," >&2
+    echo "get cmake from your package manager or compile it from source." >&2
+    exit 1;;
+esac
+
+deps/tools/jldownload https://cmake.org/files/v$CMAKE_VERSION_MAJMIN/$FULLNAME.tar.gz
+echo "$CMAKE_SHA256 $FULLNAME.tar.gz" | sha256sum -c -
+tar -xzf $FULLNAME.tar.gz
+echo "CMAKE = $PWD/$CMAKE_EXTRACTED_PATH" >> Make.user
```

Comment:
It'd be nice to auto-set `CMAKE` to our possible installation downloaded paths if they exist, since they are easily calculable, and could still be easily overwritten from `Make.user`, so we wouldn't lose any flexibility if the user for whatever reason didn't want to use our downloaded `CMAKE`.